src/pages/how-does-react-tell-a-class-from-a-function/index.nl.md

```diff
@@ -0,0 +1,553 @@
+fred.sayHi();
+```
+
+Dit is hoe mensen vroeger classes nabootsten voordat ze officieel werden toegevoegd aan JavaScript.
+
+---
+
+Waar `new` al een tijdje te gebruiken is in JavaScript, zijn classes nieuwer. Classes maken het ons mogelijk om de code hierboven te herschrijven op een leesbaardere manier die duidelijker maakt wat we ermee willen bereiken:
+
+```jsx
+class Person {
+  constructor(name) {
+    this.name = name;
+  }
+  sayHi() {
+    alert('Hi, I am ' + this.name);
+  }
+}
+
+let fred = new Person('Fred');
+fred.sayHi();
+```
+
+*Die duidelijkheid over wat de developer ergens mee wil bereiken is enorm belangrijk.* Het is dus belangrijk om hier rekening mee te houden bij zowel het ontwerpen van programmeertalen als API’s.
+
+Als je een functie schrijft kan JavaScript niet zelf inschatten of deze moet worden aangeroepen met `alert()` of dat het een constructor moet zijn zoals `new Person()`. `new` vergeten bij het aanroepen van `Person` kan voor verwarrende resultaten zorgen.
+
+**Dankzij de syntax van class is het mogelijk te zeggen: “Dit is niet zomaar een functie - het is een class en het heeft een constructor”.** Als je `new` vergeet te gebruiken wanneer je deze aanroept geeft JavaScript een error:
+
+```jsx
+let fred = new Person('Fred');
+// ✅ Als Person een functie is: werkt prima
+// ✅ Als Person een class is: werkt ook prima
+
+let george = Person('George'); // We zijn ‘new’ vergeten
+// 😳 Als Person een constructor-achtige functie is: verwarrend geïmplementeerd.
+// 🔴 Als Person een class is: geeft direct een error
+```
+
+Dit helpt ons om snel fouten te vinden. In plaats van dat we moeten wachten totdat er een rare bug tevoorschijn komt. Zoals `this.name` die geïmplementeerd wordt als `window.name` in plaats van `george.name`.
+
+Dit betekent echter wel dat React `new` moet gebruiken voordat een class aangeroepen wordt. Een class kan niet aangeroepen worden als een normale functie, sinds JavaScript dit als een error zou behandelen.
+
+```jsx
+class Counter extends React.Component {
+  render() {
+    return <p>Hello</p>;
+  }
+}
+
+// 🔴 Dit is in React niet zomaar mogelijk:
+const instance = Counter(props);
+```
+
+Dit vraagt om problemen.
+
+---
+
+Voordat we gaan kijken hoe React dit oplost is het belangrijk om te realiseren dat de meeste gebruikers van React compilers zoals Babel gebruiken om moderne functionaliteiten zoals classes te kunnen gebruiken in oudere browsers. Dus we moeten rekening houden met compilers in het ontwerp van React.
+
+In oudere versies van Babel konden classes aangeroepen worden zonder `new`. Dit is echter opgelost  - door middel van wat extra code.
+
+```jsx
+function Person(name) {
+  // Een versimpelde versie van de Babel output:
+  if (!(this instanceof Person)) {
+    throw new TypeError("Cannot call a class as a function");
+  }
+  // Onze code:
+  this.name = name;
+}
+
+new Person('Fred'); // ✅ Okay
+Person('George'); // 🔴 Een class kan je niet aanroepen als een functie.
```

Comment:
Should we translate this line? I'm asking since line 193 where the error is defined hasn't been translated either.

Reply:
Agreed!